Sources/NotionSwift/Models/PageProperty.swift

```diff
@@ -239,8 +239,9 @@ extension PagePropertyType: Codable {
                 String.self,
                 forKey: .url
             )
-            if let value = value {
-                self = .url(URL(string: value))
+            if let value = value,
+               let url = URL(string: value) {
```

Comment:
Ah right, I now see that it doesn't make any sense, thanks a lot for pointing it out. I tricked myself into thinking it was a fix, as I couldn't reproduce the issue with it. Now I also cannot reproduce it without this 'fix'. I'm seeing this issues in my analytics tool (multiple users over last 7 days):

`decodingError(Swift.DecodingError.valueNotFound(Swift.String, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "properties", intValue: nil), _JSONKey(stringValue: "URL", intValue: nil), CodingKeys(stringValue: "url", intValue: nil)]`

I was able to reproduce it before by saving a page and not specifying the url property (while the url property was configured for the parent database). Now it all works fine, which is bittersweet. I'll keep digging and will let you know 👌🏻

Reply:
maybe in you app you're using older SDK version (without `.url` supporting optional value)?

Reply:
That might be the case - I've bumped to the latest version and haven't seen this issue since. I've removed above change and updated the ticket, thanks a lot for helping out 🙌🏻